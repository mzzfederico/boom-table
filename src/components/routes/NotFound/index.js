import React from "react";
import {withRouter} from "react-router-dom";

const NotFound = ({
    location
}) => (
    <React.Fragment>
        <h1>Not found</h1>
        <h2>Nothing found for page {location.pathname}.</h2>
    </React.Fragment>
);

export default withRouter(NotFound);