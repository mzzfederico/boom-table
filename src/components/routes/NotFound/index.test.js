/* eslint-disable no-undef */
import React from "react";
import NotFound from "./index";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[ "/botched_route" ]}>
            <NotFound />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it("displays the bad route", () => {
    const tree = renderer.create(
        <MemoryRouter initialEntries={[ "/botched_route" ]}>
            <NotFound />
        </MemoryRouter>
    ).toJSON();
    expect(tree.indexOf("/botched_route") > -1);
});
