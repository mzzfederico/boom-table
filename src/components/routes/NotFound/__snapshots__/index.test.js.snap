// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <h1>
    Not found
  </h1>,
  <h2>
    Nothing found for page 
    /botched_route
    .
  </h2>,
]
`;
